### PR TITLE
Using PowerShell in Packages and Scripts

### PR DESCRIPTION
**Pull request by thenley**
New article

### CODE REVIEW

File: Blueprints/using-powershell-in-packages-and-scripts.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+* [Building Windows 2012 Script Packages](https://www.centurylinkcloud.com/knowledge-base/blueprints/building-windows-2012-script-packages/)
+
+
+### Overview
+
+When creating packages and scripts, there are many different options.  This article offers some guidance on using PowerShell as your scripting language.
+
+* Be sure to set the execution mode to PowerShell in the package.manifest.
+* Avoid use of console commands, such as Write-Host.  These will cause your scripts to fail when run in a script or package.
+* Scripts are executed from a temporary directory.  Any files you output, such as logs should be written to a directory in another directory.  Alternatively, you can set a flag in the package to [make the directory permanent](https://www.centurylinkcloud.com/knowledge-base/blueprints/blueprints-script-and-software-package-management/).
```
Comment (osbornm) [inline Blueprints/using-powershell-in-packages-and-scripts.md]: when linking to other article in the KB us relative links please. in this case `blueprints-script-and-software-package-management.md` as the link and our import process will fix them up. 

https://github.com/centurylinkcloud/PublicKB#links-kb-article-to-kb-article

File: Blueprints/using-powershell-in-packages-and-scripts.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* The built-in methods to encrypt passwords and data does not typically work as the key changes for each system.
+* Specify the PowerShell version required in the script to ensure it only runs if that version is present.  This is particularly important if you are using the API with the Invoke-RestMethod as it requires version 3.
+* Use error trapping and logging to determine if the script executed properly.  
+* Using "exit 0" in a script will tell the platform it was successful.  Any other exit codes are considered failures.
```
Comment (osbornm) [inline Blueprints/using-powershell-in-packages-and-scripts.md]: try using the " back tick " character for when displaying input or out put it `makes it stand out like this`

**Comment by chrislittle** [comment]
Ted, you may want to check out my template @ https://ca.tier3.io/filestore/template.md 

it goes over some of this stuff so you can be compliant.

**Comment by thenley** [comment]
Thanks.  I will!

Ted Henley | Cloud Architect

CenturyLink Cloud
1100 112th Avenue NE, Suite 400, Bellevue, WA 98004 
Mobile: 630-728-7775

www.centurylinkcloud.com<http://www.centurylinkcloud.com/>

**Comment by osbornm** [comment]
@chrislittle why is that a file on some file store and not in the readme? Can you make a pull request to put all that in the readme? or at very least the github wiki :) that would be helpful

**Comment by nathanyoung** [comment]
Closing due to inactivity. Feel free to resubmit after changes are made.

**Pull request closed without merging**